### PR TITLE
[Core] Remove Python grpcio from check_health

### PR DESCRIPTION
**Pull request by pcmoritz**
## Why are these changes needed?

Next step in https://github.com/ray-project/ray/issues/35472

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/gcs_client/gcs_client.cc [hunk 1]
```diff
@@ -408,5 +408,41 @@ std::unordered_map<std::string, std::string> PythonGetNodeLabels(
                                                       node_info.labels().end());
 }
 
+Status PythonCheckGcsHealth(const std::string &gcs_address,
+                            int gcs_port,
```
Comment (vitsai) [inline src/ray/gcs/gcs_client/gcs_client.cc:412]: nit: const

File: src/ray/gcs/gcs_client/gcs_client.cc [hunk 2]
```diff
@@ -408,5 +408,41 @@ std::unordered_map<std::string, std::string> PythonGetNodeLabels(
...
+                            int64_t timeout_ms,
```
Comment (vitsai) [inline src/ray/gcs/gcs_client/gcs_client.cc:413]: nit: const

File: src/ray/gcs/gcs_client/gcs_client.cc [hunk 3]
```diff
@@ -408,5 +408,41 @@ std::unordered_map<std::string, std::string> PythonGetNodeLabels(
...
+                            const std::string &ray_version,
+                            bool skip_version_check,
```
Comment (vitsai) [inline src/ray/gcs/gcs_client/gcs_client.cc:415]: nit: const

File: python/ray/_raylet.pyx [hunk 4]
```diff
@@ -2622,6 +2623,42 @@ cdef class _TestOnly_GcsActorSubscriber(_GcsSubscriber):
...
+        c_string c_gcs_address = gcs_address
+        int c_gcs_port = int(gcs_port)
+        int64_t timeout_ms = round(1000 * timeout) if timeout else -1
+        c_string c_ray_version = ray.__version__
+        c_bool c_skip_version_check = skip_version_check
+        c_bool c_is_healthy = True
+
+    try:
+        with nogil:
+            check_status(PythonCheckGcsHealth(
```
Comment (vitsai) [inline python/ray/_raylet.pyx:2653]: I think the behavior here is different from before? If it was RPC error before, it returned false.

**Review by vitsai — APPROVED** [review]
Conditioned on tests

File: python/ray/_raylet.pyx (hunk 4 above)
Comment (pcmoritz) [reply python/ray/_raylet.pyx:2653]: You are right, I need to catch `RpcError` here and in that case ignore it :)

File: src/ray/gcs/gcs_client/gcs_client.cc (hunk 2 above)
Comment (pcmoritz) [reply src/ray/gcs/gcs_client/gcs_client.cc]: fixed

File: src/ray/gcs/gcs_client/gcs_client.cc (hunk 3 above)
Comment (pcmoritz) [reply src/ray/gcs/gcs_client/gcs_client.cc]: fixed

File: src/ray/gcs/gcs_client/gcs_client.cc (hunk 1 above)
Comment (pcmoritz) [reply src/ray/gcs/gcs_client/gcs_client.cc]: fixed

File: python/ray/_raylet.pyx (hunk 4 above)
Comment (pcmoritz) [reply python/ray/_raylet.pyx:2653]: fixed now

**Comment by vitsai** [comment]
LGTM, looks like you forgot to sign your commit

**Pull request merged**